recipes/libsolace/all/conanfile.py

```diff
@@ -0,0 +1,62 @@
+import os
+
+from conans import CMake, ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+
+class LibsolaceConan(ConanFile):
+    name = "libsolace"
+    description = "High performance components for mission critical applications"
+    topics = ("HPC", "High reliability", "P10", "solace", "performance", "c++", "conan")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/abbyssoul/libsolace"
+    license = "Apache-2.0"
+    settings = "os", "compiler", "build_type", "arch"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False]
+    }
+    default_options = "shared=False", "fPIC=True"
+    generators = "cmake"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _supported_cppstd(self):
+        return ["17", "gnu17", "20", "gnu20"]
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os == "Windows":
+          raise ConanInvalidConfiguration("This library is not yet compatible with Windows")
+        if self.settings.compiler.cppstd and not self.settings.compiler.cppstd in self._supported_cppstd:
+          raise ConanInvalidConfiguration("This library requires c++17 standard or higher. {} required".format(self.settings.compiler.cppstd))
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = self.name + "-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self, parallel=True)
+        cmake.definitions["BUILD_TESTING"] = "OFF"
+        cmake.definitions["PKG_CONFIG"] = "OFF"
+        cmake.definitions["SOLACE_GTEST_SUPPORT"] = "OFF"
+        cmake.configure(source_folder=self._source_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+
+    def package_info(self):
+        self.cpp_info.libs = ["solace"]
```

Comment:
```suggestion
        self.cpp_info.libs = ["solace"]
        if self.settings.os == "Linux":
            self.cpp_info.libs.append("m")
```

According ldd, it requires libm:
```
ldd /home/uilian/.conan/data/libsolace/0.3.6/_/_/package/7f2f2e0dc409417fb8c1337f4abdc3c40fe90b3e/lib/libsolace.so
	linux-vdso.so.1 (0x00007ffc3472f000)
	libstdc++.so.6 => /usr/lib/libstdc++.so.6 (0x00007f8dae974000)
	libm.so.6 => /usr/lib/libm.so.6 (0x00007f8dae82e000)
	libgcc_s.so.1 => /usr/lib/libgcc_s.so.1 (0x00007f8dae814000)
	libc.so.6 => /usr/lib/libc.so.6 (0x00007f8dae64d000)
	/usr/lib64/ld-linux-x86-64.so.2 (0x00007f8daebc7000)
```

Reply:
Oh - so that's how you do it. All my projects that consume `libsolace/abbyssoul` do depend on `libm` so that's why it was not an issue. 
Thanks.